File: anthropic/clients/python/test_connection.py
"""Test Anthropic Claude Python client connection"""
from anthropic_client import AnthropicClient


def main():
    print('\n🔌 Anthropic Claude Python Connection Test\n')
    client = AnthropicClient()
    passed = failed = 0

    def test(name, fn):
        nonlocal passed, failed
        try:
            fn()
            print(f'  ✅ {name}')
            passed += 1
        except Exception as e:
            print(f'  ❌ {name}: {e}')
            failed += 1

    def t_chat():
        res = client.generate_content(
            'Say "Claude connected" — nothing else.',
            model='claude-haiku-4-5',
            max_tokens=10
        )
        assert 'claude' in res['text'].lower() or 'connected' in res['text'].lower()

    def t_streaming():
        chunks = list(client.generate_content_stream(
            'Count to 3.', model='claude-sonnet-4-6', max_tokens=20
        ))
        text = ''.join(c['text'] for c in chunks if not c['done'])
        assert text, 'No streaming output'

    test('Chat (claude-haiku-4-5)', t_chat)
    test('Streaming (claude-sonnet-4-6)', t_streaming)

    print(f'\n  {passed} passed · {failed} failed')
    exit(1 if failed else 0)


if __name__ == '__main__':
    main()

File: anthropic/clients/node/test-connection.js
const { AnthropicClient } = require('./anthropic-client');

async function main() {
  console.log('\n🔌 Anthropic Claude Connection Test\n');
  const client = new AnthropicClient();
  let passed = 0, failed = 0;

  async function test(name, fn) {
    try { await fn(); console.log(`  ✅ ${name}`); passed++; }
    catch (e) { console.log(`  ❌ ${name}: ${e.message || JSON.stringify(e)}`); failed++; }
  }

  await test('Chat (claude-haiku-4-5)', async () => {
    const res = await client.generateContent({
      prompt: 'Say "Claude connected" — nothing else.',
      model: 'claude-haiku-4-5',
      max_tokens: 10
    });
    if (!res.text.toLowerCase().includes('claude') && !res.text.toLowerCase().includes('connected'))
      throw new Error(`Got: ${res.text}`);
  });

  await test('Streaming (claude-sonnet-4-6)', async () => {
    let text = '';
    for await (const chunk of client.generateContentStream({
      prompt: 'Count to 3.',
      model: 'claude-sonnet-4-6',
      max_tokens: 20
    })) { text += chunk.text; }
    if (!text) throw new Error('No output');
  });

  console.log(`\n  ${passed} passed · ${failed} failed`);
  process.exit(failed > 0 ? 1 : 0);
}

main().catch(e => { console.error(e); process.exit(1); });
